astropy/coordinates/representation.py

```diff
@@ -538,12 +773,24 @@ def _combine_operation(self, op, other, reverse=False):
             Whether the operands should be reversed (e.g., as we got here via
             ``self.__rsub__`` because ``self`` is a subclass of ``other``).
         """
+        self._raise_if_has_differentials(op.__name__)
+
         result = self.to_cartesian()._combine_operation(op, other, reverse)
         if result is NotImplemented:
             return NotImplemented
         else:
             return self.from_cartesian(result)
 
+    # We need to override this setter to support differentials
+    @BaseRepresentationOrDifferential.shape.setter
+    def shape(self, shape):
+        # See: https://stackoverflow.com/questions/3336767/ for an example
+        BaseRepresentationOrDifferential.shape.fset(self, shape)
```

Comment:
Not sure about the first part: so far, we require that the differentials have the same shape as the representation, so I don't think we'll have broadcasting issues? Of course, it would be nice to allow them to just be broadcastable...

`super()` doesn't work, which is why I had to resort to the class name...not sure if that is the best way to do it though.